### ISSUE TITLE
[esbuild-plugin-html] Wrong script src when using sourcemap

### ISSUE THREAD

**Issue by junkey1000** [chialab/rna#45]
I have a problem when using the `esbuild-plugin-html` (0.15.9) with `esbuild` option `sourcemap` enabled.
This results in as html file that references the `.js.map` file instead of the desired `.js` file.

Excerpt of `esbuild.config.js` file:
```
import esbuild from 'esbuild';
import htmlPlugin from '@chialab/esbuild-plugin-html';

esbuild.build({
	entryPoints: [ './index.html' ],
	plugins: [ htmlPlugin() ],
	bundle: true,
	sourcemap: true,
});
```
Content of the `index.html` file:
```
<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8">
	<script src="./index.ts"></script>
</head>
<body>
	<div id="app"></div>
</body>
</html>
```
Excerpt of output `index.html` file:
`<script src="1.js.map" type="application/javascript"></script>`
Desired output:
`<script src="1.js" type="application/javascript"></script>`

Possible fix in file `esbuild-plugin-html/lib/collectScripts.js`
+ line 41:
`const jsOutput = files.find((file) => file.endsWith('.js'));`
+ line 52:
`const cssOutputs = files.filter((file) => file.endsWith('.css'));`

**Comment by edoardocavazza** [comment]
Hello @junkey1000, thanks for reporting. We released a new version of the plugin (0.15.17) that should fix the issue. Can you check if the proposed solution works for you? Thank you!

**Comment by eco747** [comment]
Hi,
Not working for me on version 0.15.37

with sourcemap: false
```html
<body>
    <script src="js\2-N7R55QZ6.js" type="module"></script>
</body>
```

with sourcesmap: "external"
```html
<body>
    <script src="js\2-GNKWUMEG.js.map" type="module"></script>
</body>
```
